src/modules/roc_sndio/DriverInfo.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_sndio/DriverInfo.h
+//! @brief Driver info interface.
+
+#ifndef ROC_SNDIO_DRIVERINFO_H_
+#define ROC_SNDIO_DRIVERINFO_H_
+
+namespace roc {
+namespace sndio {
+
+//! Driver info interface.
+class DriverInfo {
+public:
+    DriverInfo();
+    DriverInfo(const char* name)
+        : name(name){};
+    const char* name;
```

Comment:
```
const char* name;
```

This assumes that the name is a constant string which lifetime is not limited. While this is true for current backends, this looks a bit dangerous because these strings come from external libraries. So I'd prefer to avoid this assumption here and use something like `char name[16]` here.

Reply:
I'm not sure I follow, while external libraries can store strings as constant char* or char[], wouldnt it be up to each backend's implementation of get_drivers to resolve this?  This is the reason I defined it as an abstract function.  Or is it just more convenient to store it as an array?

Reply:
If the external library stores names as a static constant string (which lifetime is not limited), we can go both ways: store the name in `char*` or copy it to `char[N]`.

But if the external library provides a temporary string which lifetime ends after get_drivers() returns, we can't just store the pointer in `char*` because it will become invalid after get_driver() returns. So in this case we *should* copy it to `char[N]`.

So the second approach (`char[N]`) is more universal: it makes no assumptions about the lifetime of the strings provided by backend.